pkg/resource/trail/custom_update_api.go

```diff
@@ -0,0 +1,145 @@
+package trail
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/cloudtrail"
+)
+
+// customUpdateTrail implements a custom logic for handling Trail
+// resource updates.
+func (rm *resourceManager) customUpdateTrail(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateTrail")
+	defer exit(err)
+
+	var updated *resource
+	updated = desired
+	if delta.DifferentAt("Spec.CloudWatchLogsLogGroupARN") ||
+		delta.DifferentAt("Spec.CloudWatchLogsRoleARN") ||
+		delta.DifferentAt("Spec.EnableLogFileValidation") ||
+		delta.DifferentAt("Spec.IncludeGlobalServiceEvents") ||
+		delta.DifferentAt("Spec.IsMultiRegionTrail") ||
+		delta.DifferentAt("Spec.IsOrganizationTrail") ||
+		delta.DifferentAt("Spec.KMSKeyID") ||
+		delta.DifferentAt("Spec.S3BucketName") ||
+		delta.DifferentAt("Spec.S3KeyPrefix") ||
+		delta.DifferentAt("Spec.SnsTopicName") {
+		err = rm.updateTrailField(ctx, updated)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if delta.DifferentAt("Spec.Tags") {
+		err = rm.syncTrailTags(ctx, latest, updated)
+		if err != nil {
+			return nil, err
+		}
+	}
+	readOneLatest, err := rm.ReadOne(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+	return rm.concreteResource(readOneLatest), nil
+}
+
+// syncTrailTags updates a trail list of tags.
+func (rm *resourceManager) syncTrailTags(
+	ctx context.Context,
+	latest *resource,
+	desired *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTrailTags")
+	defer exit(err)
+
+	added, updated, removed := computeTagsDelta(latest.ko.Spec.Tags, desired.ko.Spec.Tags)
```

Comment:
Note that there's really no need to track "updated" separately from "added" because, according to [the docs](https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_AddTags.html) "Overwrites an existing tag's value when a new value is specified for an existing tag key"